packages/langium/src/workspace/document-builder.ts

```diff
@@ -157,10 +157,7 @@ export class DefaultDocumentBuilder implements DocumentBuilder {
         this.indexManager.remove(deleted);
         // Set the state of all changed documents to `Changed` so they are completely rebuilt
         for (const changedUri of changed) {
-            const invalidated = this.langiumDocuments.invalidateDocument(changedUri);
-            if (!invalidated) {
-                this.langiumDocuments.getOrCreateDocument(changedUri);
```

Comment:
This line was important in order to load documents that are new (the name `changed` is misleading in this regard: it includes new documents). I confirmed that the change introduces a bug: when I create a new file, it's not considered by the document builder.

We have to load these new documents before the first call to `interruptAndCheck` to ensure that we don't lose them.

Idea: create "empty" documents with state `Changed` and no AST (this can be done synchronously). The document builder will parse the actual content in step 0 of `buildDocuments`.

Reply:
I've kept the same logic. The contract of our `LangiumDocument` doesn't allow for an `undefined` parse value, so it didn't feel right to create an empty document. We now parse new documents in parallel though.

Reply:
I'm thinking about changing the contract. The problem is that the initial parsing is done during the non-interruptible phase. The same applies to the WorkspaceManager: we have to parse the whole workspace before we pass it to the DocumentBuilder.

Making `parseResult` optional could be a quite heavy (and inconvenient) change. WDYT of creating a synthetic root node as placeholder?